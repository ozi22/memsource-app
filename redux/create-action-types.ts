export const createActionTypes = (actionTypes: any = {}, duckName = 'DUCK'): any => {
  return Object.keys(actionTypes).reduce(
    (res, key) => ({
      ...res,
      [key]: `${duckName.toUpperCase()}_DUCK/${actionTypes[key].toUpperCase()}`
    }),
    {}
  );
};
