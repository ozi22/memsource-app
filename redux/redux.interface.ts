import { ThunkAction } from 'redux-thunk';

export interface IAction<Payload = any, Opt = any> {
  type: string;
  payload: Payload;
  opt?: Opt;
}

export type ThunkResult<R> = ThunkAction<R, any, undefined, any>;
