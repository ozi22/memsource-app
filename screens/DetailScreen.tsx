import React, { FC } from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { useSelector } from 'react-redux';
import ProjectsDuck from '../redux/projects/projects.duck';
import Colors, { statuses } from '../constants/Colors';

const DetailScreen: FC = () => {
  const { name, sourceLang, targetLangs, status } = useSelector(ProjectsDuck.getDetail);
  return (
    <View style={styles.container}>
      <View>
        <Text style={styles.itemName}>{name}</Text>
        <View style={{ flexDirection: 'row' }}>
          <Text style={styles.lang}>
            {sourceLang}
            {` > `}
            {targetLangs?.map((item, index, { length }) => `${item}${index !== length - 1 ? ', ' : ''}`)}
          </Text>
        </View>
      </View>
      <Text style={[styles.status, { backgroundColor: statuses[status] }]}>{status}</Text>
    </View>
  );
}

export default DetailScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 10,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: '#dcdee2',
    backgroundColor: '#fff',
    padding: 15
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%'
  },
  itemName: {
    fontWeight: 'bold',
    color: Colors.light.text,
    fontSize: 16,
    paddingBottom: 3
  },
  lang: {
    color: Colors.light.tabIconDefault,
    display: 'flex',
    alignItems: 'center'
  },
  status: {
    padding: 10,
    alignSelf: 'flex-start'
  }
});
