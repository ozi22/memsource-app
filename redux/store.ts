import { applyMiddleware, combineReducers, createStore } from 'redux';
import reduxThunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import UIDuck from './ui/ui.duck';
import LoginDuck from './login/login.duck';
import ProjectsDuck from './projects/projects.duck';

const combineDucks = (...ducks: any): any =>
  combineReducers(
    ducks.reduce((root: any, duck: any) => {
      return { ...root, [duck.name]: duck.reducer };
    }, {})
  );

export type AppState = ReturnType<typeof combineDucks>;

const rootReducer: any = combineDucks(UIDuck, LoginDuck, ProjectsDuck);

export const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(reduxThunk)));
