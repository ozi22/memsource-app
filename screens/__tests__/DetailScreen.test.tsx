import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import DetailScreen from '../DetailScreen';

describe('DetailScreen', () => {
  it('should render correctly', () => {
    const { toJSON } = renderer.create(
      <Provider store={store}>
        <DetailScreen />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
