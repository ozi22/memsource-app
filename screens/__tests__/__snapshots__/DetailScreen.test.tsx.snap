// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen should render correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#dcdee2",
      "borderStyle": "solid",
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "padding": 15,
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#242d37",
          "fontSize": 16,
          "fontWeight": "bold",
          "paddingBottom": 3,
        }
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "center",
            "color": "#435569",
            "display": "flex",
          }
        }
      >
         &gt; 
      </Text>
    </View>
  </View>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "padding": 10,
        },
        Object {
          "backgroundColor": undefined,
        },
      ]
    }
  />
</View>
`;
